src/Runner/TestInstance.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/**
+ * This file is part of the Nette Tester.
+ * Copyright (c) 2009 David Grudl (https://davidgrudl.com)
+ */
+
+namespace Tester\Runner;
+
+
+/**
+ * Carries information about a single test instance.
+ */
+class TestInstance
+{
+
+	/** @var Job|NULL */
+	private $job;
+
+	/** @var string */
+	private $fileName;
+
+	/** @var string */
+	private $testName;
+
+	/** @var string|NULL */
+	private $instanceName;
+
+	/** @var int */
+	private $result;
+
+	/** @var string|NULL */
+	private $message;
+
+
+	private function __construct()
+	{
+	}
+
+
+	/**
+	 * @param Job $job
+	 * @param string $testName
+	 * @return TestInstance
+	 */
+	public static function withJob(Job $job, $testName)
+	{
+		$instance = new self();
+		$instance->job = $job;
+		$instance->fileName = $job->getFile();
+		$instance->testName = $testName;
+
+		$args = $job->getArguments();
+		$instanceNameParts = [];
+		if ($args && isset($args['method'])) {
+			$instanceNameParts[] = $args['method'];
+		}
+
+		if (isset($args['multiple'])) {
+			$instanceNameParts[] = '#' . $args['multiple'];
+		}
+
+		if (isset($args['dataprovider'])) {
+			$instanceNameParts[] = '(' . $args['dataprovider'] . ')';
+		}
+
```

Comment:
These `method`, `multiple`, `dataprovider` references smell to me a little bit. Not sure how to do it. Maybe the `TestInstance` is now better place for homehow pass them to job.